include/event2/ws.h

```diff
@@ -0,0 +1,56 @@
+#ifndef EVENT2_WS_H_INCLUDED_
+#define EVENT2_WS_H_INCLUDED_
+
+struct evws_connection;
+
+#define WS_CR_NONE 0
+#define WS_CR_NORMAL 1000
+#define WS_CR_PROTO_ERR 1002
+#define WS_CR_DATA_TOO_BIG 1009
+
+#define WS_TEXT_FRAME 0x1
+#define WS_BINARY_FRAME 0x2
+
+typedef void (*ws_on_msg_cb)(
+	struct evws_connection *, int type, struct evbuffer *, size_t, void *);
+typedef void (*ws_on_close_cb)(struct evws_connection *, void *);
+
+/** Opens new WebSocket session from HTTP request.
+ *
+ * @param req a request object
+ * @param cb the callback function that gets invoked on receiving message
```

Comment:
Let's describe the behavior of the buffer, i.e. that user should not modify it, and libevent will remove processed bytes by itself.